### PR TITLE
Format README

### PR DESCRIPTION
**Pull request by knjk04**
### Problem

The README didn't have the .md file extension, and so it was harder to read than necessary.

### Solution

I added the file extension and formatted the markdown (e.g. added syntax highlighting and headings) to make the README easier to read.

### Result

This doesn't affect the end user but the programmers. The changes make it a lot nicer to read the README.

### CODE REVIEW

File: src/java/com/twitter/common/styleguide.md [hunk 1]
```diff
@@ -13,7 +13,38 @@ In general, much of our style and conventions mirror the
 [Code Conventions for the Java Programming Language](http://www.oracle.com/technetwork/java/codeconvtoc-136057.html)
 and [Google's Java Style Guide](https://google.github.io/styleguide/javaguide.html).
 
-[TOC]
+
+## Table of contents
+
+- [Recommended reading](https://github.com/knjk04/commons/blob/master/src/java/com/twitter/common/styleguide.md#recommended-reading)
```
Comment (Yasumoto) [inline src/java/com/twitter/common/styleguide.md]: Sweet! How’d you generate this? Mind replacing the `knjk04` with `twitter`? 🙏

**Comment by knjk04** [comment]
I typed this out by hand! Yeah, sure. Sorry, I didn't realise I was getting the links from my for :)

File: src/java/com/twitter/common/styleguide.md (hunk 1 above)
Comment (knjk04) [reply src/java/com/twitter/common/styleguide.md]: I typed this out by hand! Yeah, sure. Sorry, I didn't realise I was getting the links from my for :)

I can look into ways to generate the TOC. It's probably easier for everyone else.

Comment (knjk04) [reply src/java/com/twitter/common/styleguide.md]: Update: I've made the changes you asked for in a different branch. See PR: https://github.com/twitter/commons/pull/467

**Comment by grimreaper** [comment]
Thank you for your contribution. Unfortunately, we’re not going to continue maintaining twitter-commons and are archiving all pull requests.

**Pull request closed without merging**